index.ts

```diff
@@ -203,6 +206,27 @@ router.get("/discord", async (ctx) => {
     ctx.redirect("https://discord.gg/gT9ZZDqTyx");
 });
 
+const mongo = new Stores.Mongodb(mongoose.connection, {
+    collectionName: 'ratelimits',
+    collectionAbuseName: 'ratelimitsabuses',
+});
+
+export const rateLimiter = RateLimit.middleware({
+    interval: { sec: 1 },  // 1 second
+    max: async (ctx) => {
+        const apiKey = ctx.query.key;
+        const rateInfo = await retrieveRateLimit(apiKey);
+        return rateInfo.rateLimit; // max requests per second per team
+    },
+    keyGenerator: async (ctx) => {
+        const apiKey = ctx.query.key;
+        const rateInfo = await retrieveRateLimit(apiKey);
+        return `teamid:${rateInfo.teamNumber}`; // team number as key
+    },
+    store: mongo,
+    message: "Exceeded the rate limit. Try again later."
+});
+
```

Comment:
would be good if we can move this stuff to a file in the middleware folder since it's acting like middleware (also this way the imports are more intuitive/readable from the router)